src/main/scala/io/iohk/ethereum/ledger/Ledger.scala

```diff
@@ -2,78 +2,71 @@ package io.iohk.ethereum.ledger
 
 import akka.util.ByteString
 import io.iohk.ethereum.consensus.Consensus
-import io.iohk.ethereum.consensus.validators.BlockHeaderError.HeaderParentNotFoundError
 import io.iohk.ethereum.domain._
 import io.iohk.ethereum.ledger.BlockExecutionError.ValidationBeforeExecError
-import io.iohk.ethereum.ledger.BlockQueue.Leaf
 import io.iohk.ethereum.ledger.Ledger._
-import io.iohk.ethereum.metrics.{Metrics, MetricsClient}
 import io.iohk.ethereum.utils.Config.SyncConfig
-import io.iohk.ethereum.utils.{BlockchainConfig, DaoForkConfig, Logger}
+import io.iohk.ethereum.utils.{ BlockchainConfig, Logger }
 import io.iohk.ethereum.vm._
-import org.bouncycastle.util.encoders.Hex
 
-import scala.annotation.tailrec
-import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.{ ExecutionContext, Future }
 
 trait Ledger {
   def consensus: Consensus
 
   def checkBlockStatus(blockHash: ByteString): BlockStatus
 
-  /**
-   * Executes a block
-   *
-   * @param alreadyValidated should we skip pre-execution validation (if the block has already been validated,
-   *                         eg. in the importBlock method)
-   */
+  /** Executes a block
+    *
+    * @param alreadyValidated should we skip pre-execution validation (if the block has already been validated,
+    *                         eg. in the importBlock method)
+    */
   def executeBlock(block: Block, alreadyValidated: Boolean = false): Either[BlockExecutionError, Seq[Receipt]]
 
   def simulateTransaction(stx: SignedTransactionWithSender, blockHeader: BlockHeader, world: Option[InMemoryWorldStateProxy]): TxResult
```

Comment:
food for thought: Maybe it would be good to move `simulateTransaction` and `binarySearchGasEstimation` to separate class ? They are highly connected to each other, but not have much in common with other methods. As i see Ledger public interface it should be highly related to importing block to blockchain and providing info about the process, so from current interface the most releveant methods are: 
`checkBlockStatus`
`importBlock`
`resolveBranch`